demo/runtime/src/lib.rs

```diff
@@ -174,17 +178,17 @@ impl_outer_event! {
 }
 
 impl_outer_log! {
-	pub enum Log for Runtime {
-		consensus
+	pub enum Log(InternalLog: DigestItem<SessionKey>) for Runtime {
+		consensus(AuthoritiesChange)
 	}
 }
 
 impl DigestItem for Log {
-	type AuthoritiesChange = consensus::AuthoritiesChange<SessionKey>;
+	type AuthorityId = SessionKey;
 
-	fn as_authorities_change(&self) -> Option<&Self::AuthoritiesChange> {
-		match *self {
-			Log::consensus(ref item) => item.as_authorities_change(),
+	fn as_authorities_change(&self) -> Option<&[Self::AuthorityId]> {
```

Comment:
can't we just re-use the `DigestItem` as defined in `substrate-runtime-primitives::traits`?

Reply:
Not sure I got it :/ `traits::DigestItem::as_authorities_change` always returns `None` - so this won't work. If you mean `generic::DigestItem` - I suppose we could use it, but since it is a demo runtime, imo it is better to demonstrate correct impl && use `impl_outer_log` to avoid manual encoding/decoding of all non-system digest items.